serde-generate/runtime/dart/serde/binary_serializer.dart

```diff
@@ -0,0 +1,134 @@
+part of serde;
+
+abstract class BinarySerializer {
+  final List<int> output = List<int>.empty(growable: true);
+
+  Uint8List get bytes {
+    return Uint8List.fromList(output);
+  }
+
+  int get offset {
+    return output.length;
+  }
+
+  void serializeUint8List(Uint8List val) {
+    serializeLength(val.length);
+    output.addAll(val);
+  }
+
+  void serializeBytes(Bytes val) {
+    serializeLength(val.content.length);
+    output.addAll(val.content);
+  }
+
+  void serializeBool(bool val) {
+    output.addAll(Uint8List.fromList([val ? 1 : 0]));
+  }
+
+  void serializeUint8(int val) {
+    if (val.bitLength > 8) {
+      throw 'The integer literal $val can\'t be represented in 8 bits.';
+    }
+    output.addAll(Uint8List.fromList([val]));
+  }
+
+  void serializeUint16(int val) {
+    if (val.bitLength > 16) {
+      throw 'The integer literal $val can\'t be represented in 16 bits.';
+    }
+    final bdata = ByteData(2)..setUint16(0, val, Endian.little);
+    output.addAll(bdata.buffer.asUint8List());
+  }
+
+  void serializeUint32(int val) {
+    if (val.bitLength > 32) {
+      throw 'The integer literal $val can\'t be represented in 32 bits.';
+    }
+    final bdata = ByteData(4)..setUint32(0, val, Endian.little);
+    output.addAll(bdata.buffer.asUint8List());
+  }
+
+  void serializeUint64(Uint64 val) {
+    BigInt number = val.toBigInt();
+    final _byteMask = BigInt.from(0xFF);
+    int bytes = 8;
+    var bdata = Uint8List(bytes);
+    for (int i = 0; i < bytes; i++) {
+      // big endian
```

Comment:
I would rather remove commented code here and in general. (Hopefully you don't need big endian, do you?)